File: src/main/java/com/thebluehats/server/game/other/Bread.java
package com.thebluehats.server.game.other;

import net.minecraft.server.v1_8_R3.EntityPlayer;
import org.bukkit.Material;
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerItemConsumeEvent;

public class Bread implements Listener {
    @EventHandler
    public void onEat(PlayerItemConsumeEvent event) {
        if (event.getItem().getType() == Material.BREAD) {
            event.getPlayer().setHealth(Math.min(event.getPlayer().getMaxHealth(), event.getPlayer().getHealth() + 8));

            EntityPlayer player = ((CraftPlayer) event.getPlayer()).getHandle();
            player.setAbsorptionHearts(Math.min(player.getAbsorptionHearts() + 2, 4));
        }
    }
}

File: src/main/kotlin/com/thebluehats/server/game/other/Bread.kt
package com.thebluehats.server.game.other

import org.bukkit.Material
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerItemConsumeEvent

class Bread : Listener {
    @EventHandler
    fun onEat(event: PlayerItemConsumeEvent) {
        if (event.item.type == Material.BREAD) {
            event.player.health = event.player.maxHealth.coerceAtMost(event.player.health + 8)

            val player = (event.player as CraftPlayer).handle
            player.absorptionHearts = (player.absorptionHearts + 2).coerceAtMost(4f)
        }
    }
}